pkg/cluster/stores.go

```diff
@@ -34,11 +44,23 @@ func NewStoreSet(logger log.Logger, reg *prometheus.Registry, peer *Peer) *Store
 	)
 	reg.MustRegister(met)
 
+	storeNodeConnections := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "thanos_store_nodes_grpc_connections",
+		Help: "Number indicating current number of gRPC connection to store nodes. This indicates also to how many stores query node have access to.",
+	})
+	storeNodeDialHist := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "thanost_store_node_grpc_dial_seconds",
+		Help: "Histogram of block gRPC dialing latency (seconds) until connection is maintained or aborted.",
+		// Possible values: dialSuccess or dialFailure.
+	}, []string{"status"})
```

Comment:
It's usually an antipattern for metrics to distinguish success/failure through a label. Separate metric names are used for that, or in histograms, it's often not distinguished at all. Mostly because histograms are expensive enough as they are and latency doesn't matter much if failure rate is too high.

Reply:
I know, but here it's quite important to know how much time we spend blocking everything else (: on failure retries... but I guess we can drop it for now